cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -267,6 +259,7 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
                 .compose(state -> state.reconcileCas(this::dateSupplier))
                 .compose(state -> state.clusterOperatorSecret(this::dateSupplier))
                 .compose(state -> state.getKafkaClusterDescription())
+                .compose(state -> state.prepareVersionChange())
```

Comment:
Is there a reason this has to be done so early in the reconciliation?

Reply:
I think it can technically be also later on. But this is what decides if the change is feasible or not. So I think having it up top is better because it stops the upgrade / downgrade which would not work right away. Without for example already upgrading Zookeeper just to find out that Kafka cannot be upgraded. At the same time, all information is well known already here. So I did not see any reasons to do it later.

Reply:
Makes sense, thanks